src/audio/asrc/asrc.c

```diff
@@ -553,6 +557,12 @@ static int asrc_params(struct comp_dev *dev,
 	source_c = buffer_acquire(sourceb);
 	sink_c = buffer_acquire(sinkb);
 
+#if CONFIG_IPC_MAJOR_4
+	/* update the source/sink buffer formats. Sink rate will be modified below */
+	ipc4_update_buffer_format(source_c, &cd->ipc_config.base.audio_fmt);
+	ipc4_update_buffer_format(sink_c, &cd->ipc_config.base.audio_fmt);
+#endif
```

Comment:
same here - what's the effect of these changes?

Reply:
same explanation, copier doesnt set the source/sink buffer until prepare. So asrc_params( will fail without it